KubeArmor/core/hook_handlier_non_k8s.go

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of KubeArmor
+
+// Package core is responsible for initiating and maintaining interactions between external entities like K8s,CRIs and internal KubeArmor entities like eBPF Monitor and Log Feeders
+package core
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/kubearmor/KubeArmor/KubeArmor/common"
+	kl "github.com/kubearmor/KubeArmor/KubeArmor/common"
+	cfg "github.com/kubearmor/KubeArmor/KubeArmor/config"
+	"github.com/kubearmor/KubeArmor/KubeArmor/state"
+	tp "github.com/kubearmor/KubeArmor/KubeArmor/types"
+)
+
+// ListenToNonK8sHook starts listening on a UNIX socket and waits for container hooks
+// to pass new containers
+func (dm *KubeArmorDaemon) ListenToNonK8sHook() {
+	dm.Logger.Print("Started to monitor non k8s hook events")
+
+	if err := os.MkdirAll(kubearmorDir, 0750); err != nil {
+		dm.Logger.Warnf("Failed to create ka.sock dir: %v", err)
+	}
+
+	listenPath := filepath.Join(kubearmorDir, "ka.sock")
+	err := os.Remove(listenPath) // in case kubearmor crashed and the socket wasn't removed (cleaning the socket file if got crashed)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		dm.Logger.Warnf("Failed to cleanup ka.sock: %v", err)
+	}
+
+	socket, err := net.Listen("unix", listenPath)
+	if err != nil {
+		dm.Logger.Warnf("Failed listening on ka.sock: %v", err)
+		return
+	}
+
+	// Set the permissions of ka.sock to 777 so that rootless podman with user level priviledges can also communicate with the socket
+	if err := os.Chmod(listenPath, 0777); err != nil {
+		dm.Logger.Warnf("Failed to set permissions on %s: %v", listenPath, err)
+	}
+
+	defer socket.Close()
+	defer os.Remove(listenPath)
+	ready := &atomic.Bool{}
+
+	for {
```

Comment:
in case of error `socket.Accept()` as well we are trying to handle connection `dm.handleNonK8sConn()`, in that case `conn` will be `nil` and might cause panic when we're accessing it right!  

we should handle the connection only if response to the listener is successful.